app/views/account_dashboard_statuses.scala.html

```diff
@@ -0,0 +1,127 @@
+   <th scope="col">@messages("case.tabs.applications.table.status")</th>
+   <th scope="col"><span class="visuallyhidden">@messages("case.application.ruling.table.download")</span></th>
+  </tr>
+  </thead>
+  <tbody id="applications-rulings-list-body">
+  @for((c: Case, index: Int) <- cases.results.zipWithIndex) {
+  <tr id="applications-rulings-list-row-@{index}">
+   <td id="applications-rulings-list-row-@{index}-reference">@c.reference</td>
+   <td id="applications-rulings-list-row-@{index}-item_name">@c.application.goodName</td>
+   <td id="applications-rulings-list-row-@{index}-date_submitted">@Dates.format(c.createdDate)</td>
+   <td id="applications-rulings-list-row-@{index}-status">
+    @c.status match {
+    case CaseStatus.NEW => {
+     <span class="govuk-tag govuk-tag--blue">@messages("case.application.status.submitted")</span>
+    }
+    case CaseStatus.OPEN | CaseStatus.SUSPENDED => {
+     <span class="govuk-tag govuk-tag--blue">@messages("case.application.status.inProgress")</span>
+    }
+    case CaseStatus.REFERRED => {
+     <span class="govuk-tag govuk-tag--yellow">@messages("case.application.status.infoRequested")</span>
+    }
+    case CaseStatus.SUPPRESSED | CaseStatus.REJECTED => {
+     <span class="govuk-tag govuk-tag--red"> @messages("case.application.status.rejected")</span>
+    }
+    case CaseStatus.COMPLETED if c.hasActiveDecision => {
+    @if(c.daysDifference.get > 120){
+     <span class="govuk-tag govuk-tag--green"> @messages("case.application.status.approvedRuling")</span>
+    } else {
+     <span class="govuk-tag govuk-tag--red">@messages("case.application.status.approvedRulingExpiring", c.daysDifference.get)</span>
+    }
+    }
+    case CaseStatus.COMPLETED if c.hasExpiredDecision => {
+     <span class="govuk-tag govuk-tag--grey">@messages("case.application.ruling.status.expired")</span>
+    }
+    case CaseStatus.CANCELLED => {
+     <span class="govuk-tag govuk-tag--grey">@messages("case.application.status.cancelled")</span>
+    }
+    case _ => {}
+    }
+   </td>
+   <td id="applications-rulings-list-row-@{index}-download">
+    <a id="applications-rulings-list-row-@{index}-view-application-link" href="@routes.ApplicationController.viewApplication(c.reference)">
+     <span aria-hidden="true">@messages("case.application.viewApplication")</span>
+     <span class="visually-hidden">@messages("case.application.viewApplication.hidden", c.reference)</span>
+    </a>
+    <p>
+    @if(Set(CaseStatus.COMPLETED, CaseStatus.CANCELLED).contains(c.status)) {
+    <a id="applications-rulings-list-row-@{index}-view-rulings-link" href="@routes.RulingController.viewRuling(c.reference)">
+     @messages("case.application.ruling.viewRuling")
+    </a>
+    }
+    </p>
+    <p>
+     @if((c.hasActiveDecision && c.daysDifference.get <= 120) | (Set(CaseStatus.COMPLETED).contains(c.status) && c.hasExpiredDecision) | (Set(CaseStatus.CANCELLED).contains(c.status))) {
```

Comment:
What do you think about adding helper functions for things like `hasExpiringRuling`, `hasActiveRuling` etc. so that we aren't hardcoding numbers like 120 into our views?